mergify_cli/stack/cli.py

```diff
@@ -76,7 +76,13 @@ async def edit() -> None:
     "--keep-pull-request-title-and-body",
     "-k",
     is_flag=True,
-    default=asyncio.run(utils.get_default_keep_pr_title_body()),
+    # NOTE(leo): lazy loading default with `is_flag=True` is buggy in Click
+    # because it always evaluates and takes precedence on the user provided value.
+    # Use a callback instead to load the default.
+    callback=utils.get_default_keep_pr_title_body_callback,
+    # The None default allows us to lazy load the default in the callback when
+    # needed, with None meaning that the user did not provide the flag.
+    default=None,
```

Comment:
I talked with click people and to make is_flag work with dynamic default, the syntax is:

```suggestion
    flag_value=True
    default=lambda: asyncio.run(utils.get_default_keep_pr_title_body())
```

Reply:
Click introspects `default=` to set `flag_value` and when the `default` is a callable it assumes it's `True` and set `flag_value` to False`

Reply:
Indeed works fine on the example, it's leaner. I removed my workaround.